### PR TITLE
[FIX, TEST] Adds a test for read_hll_files and two fixes I encountered with the tests.

### PR DESCRIPTION
**Pull request by smehringer**
Should increase the coverage this time

### CODE REVIEW

File: test/api/union/user_bin_sequence_test.cpp [hunk 1]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
...
     EXPECT_RANGE_EQ(this->get_user_bin_kmer_counts(), (std::vector<size_t>{800, 700, 600, 500}));
 }
+
+TEST_F(user_bin_sequence_test, read_hll_files)
+{
+    size_t const k = 16;
+    size_t const b = 4;
+    hyperloglog expected(b);
+
+    std::string input_file = DATADIR"small.fa";
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:88]:
```suggestion
    std::string const input_file{data("small.fa")};
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 2]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
...
+        char* end = c_seq_it + seq.size();
+
+        while(c_seq_it + k <= end)
+        {
+            expected.add(c_seq_it, k);
+            ++c_seq_it;
+        }
+    }
+
+    this->read_hll_files(DATADIR);
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:105]:
```suggestion
    this->read_hll_files(data(""));
```
Or give the `data` implementation in `api_test.hpp` a default value:
```suggestion
    this->read_hll_files(data());
```
Or leave as is?

File: test/api/union/user_bin_sequence_test.cpp [hunk 3]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
     EXPECT_RANGE_EQ(this->get_filenames(), (std::vector<std::string>{"small2.fa", "small2.fa", "small.fa", "small.fa"}));
     EXPECT_RANGE_EQ(this->get_user_bin_kmer_counts(), (std::vector<size_t>{800, 700, 600, 500}));
 }
+
+TEST_F(user_bin_sequence_test, read_hll_files)
+{
+    size_t const k = 16;
+    size_t const b = 4;
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:85]:
```suggestion
    size_t const k{16};
    size_t const b{4};
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 4]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
...
+    hyperloglog expected(b);
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:86]:
```suggestion
    hyperloglog expected{b};
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 5]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
...
+    {
+        // we have to go C-style here for the HyperLogLog Interface
+        char* c_seq_it = &*seq.begin();
+        char* end = c_seq_it + seq.size();
+
+        while(c_seq_it + k <= end)
+        {
+            expected.add(c_seq_it, k);
+            ++c_seq_it;
+        }
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:102]:
```suggestion
        // we have to go C-style here for the HyperLogLog Interface
        const char * it = &(*seq.begin());
        char const * const end = it + seq.size() - k + 1;

        for (; it != end; ++it)
            expected.add(it, k);
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 6]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
...
+    EXPECT_EQ(this->get_sketches().size(), 4u);
+    for (auto const & sketch : this->get_sketches())
+        EXPECT_EQ(sketch.estimate(), expected.estimate());
+}
+
+TEST_F(user_bin_sequence_test, read_hll_files_empty_dir)
+{
+    seqan3::test::tmp_filename tmp_file{"some_file"};
+    auto parent = tmp_file.get_path().parent_path();
+    auto empty_dir = parent/"empty_dir";
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:116]:
```suggestion
    seqan3::test::tmp_filename const tmp_file{"some_file"};
    auto const parent_dir = tmp_file.get_path().parent_path();
    auto const empty_dir = parent_dir / "empty_dir";
```

File: test/api/union/user_bin_sequence_test.cpp [hunk 7]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
...
+    std::filesystem::create_directory(empty_dir); // create empty dir
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:117]: Can you check (on your machine, not as test) that the directory is deleted after execution?
(not sure if `tmp_filename` deletes the whole directory)

File: test/api/union/user_bin_sequence_test.cpp [hunk 8]
```diff
@@ -65,3 +78,53 @@ TEST_F(user_bin_sequence_test, sort_by_cardinalities)
...
+
+    EXPECT_THROW(this->read_hll_files(empty_dir), std::runtime_error);
+}
+
+TEST_F(user_bin_sequence_test, read_hll_files_faulty_file)
+{
+    seqan3::test::tmp_filename tmp_file{"small.hll"};
+    std::ofstream os{tmp_file.get_path()};
+    os << "I am not what a hll file looks like\n";
+    os.flush();
```
Comment (eseiler) [inline test/api/union/user_bin_sequence_test.cpp:127]:
```suggestion
    seqan3::test::tmp_filename const tmp_file{"small.hll"};
    {
        std::ofstream os{tmp_file.get_path()};
        os << "I am not what an hll file looks like\n";
    }
```

File: include/chopper/union/user_bin_sequence.hpp [hunk 9]
```diff
@@ -125,9 +125,9 @@ class user_bin_sequence
         }
         catch (std::runtime_error const & err)
         {
-            std::cerr << "[CHOPPER PACK ERROR] Something went wrong trying to read the HyperLogLog sketches from files:\n"
-                      << err.what() << '\n';
-            exit(1);
+            std::string chopper_msg{"[CHOPPER PACK ERROR] Something went wrong trying to read the HyperLogLog"
+                                    " sketches from files:\n"};
```
Comment (eseiler) [inline include/chopper/union/user_bin_sequence.hpp:129]:
```suggestion
            std::string const chopper_msg{"[CHOPPER PACK ERROR] Something went wrong trying to read the HyperLogLog"
                                          " sketches from files:\n"};
```

**Review by eseiler — CHANGES_REQUESTED** [review]

File: test/api/union/user_bin_sequence_test.cpp (hunk 7 above)
Comment (smehringer) [reply test/api/union/user_bin_sequence_test.cpp:114]: yes I checked, nothing is remaining in my tmp dir

**Review by eseiler — APPROVED** [review]

**Pull request merged**